src/trustRegion.jl

```diff
@@ -94,10 +95,33 @@ struct TrustRegion{CS, AD, FDT, L, P, ST, CJ, MTR} <:
     max_shrink_times::Int
 end
 
+struct RadiusUpdate{B}
+    simple::B
+    hei::B
+    yuan::B
+    bastin::B
+end
+
+function RadiusUpdate(;simple::Bool = Val{true}(),  # 3 different radius update schemes
+                      hei::Bool = Val{false}(),
+                      yuan::Bool = Val{false}(),
+                      bastin::Bool = Val{false}())
+    if simple
+      return RadiusUpdate{Bool}(true, false, false, false)
+    elseif hei
+      return RadiusUpdate{Bool}(false, true, false, false)
+    elseif yuan
```

Comment:
This doesn't make sense. If it's going to be non-dispatching logic, use an EnumX.

Reply:
Oh yes, I am sorry I misinterpreted the use of `Val`, but now it's clear. I have changed the overall approach.